### ISSUE TITLE
Nordpool sensor stopped working during the weekend. "Error adding entities for domain sensor with platform nordpool"

### ISSUE THREAD

**Issue by kjbu** [custom-components/nordpool#144]
Nordpool sensor stopped working during the weekend, with no changes to the system before it happened.
Started with v.0.0.4 and HA 2022.4.6. Tried to update to 0.0.5 and HA 2022.4.7 -> Same problem. 

See log in bottom.

-->

## Version of the custom_component
0.0.5

## Homeassistant version
2022.4.7

## Configuration
nordpool:
sensor:
  - platform: nordpool

    VAT: True
    currency: "NOK"
    low_price_cutoff: 0.95
    region: "Oslo"
    precision: 2
    price_type: kWh
    additional_costs: '{% set s = {
        "hourly_fixed_cost": 0.0,
        "winter_night": 0.45,
        "winter_day": 0.45,
        "summer_day": 0.45,
        "summer_night": 0.45,
        "cert": 0.01
}
%}
{% if now().month >= 5 and now().month <11 %}
    {% if now().hour >=6 and now().hour <23 %}
        {{s.summer_day+s.hourly_fixed_cost+s.cert|float}}
    {% else %}
        {{s.night+s.hourly_fixed_cost+s.cert|float}}
    {% endif %}
{% else %}
    {% if now().hour >=6 and now().hour <23 %}
        {{s.winter_day+s.hourly_fixed_cost+s.cert|float}}
    {%else%}
        {{s.winter_night+s.hourly_fixed_cost+s.cert|float}}
    {% endif %}
{% endif %}'

## Debug log

Logger: homeassistant.components.sensor
Source: helpers/template.py:409
Integration: Sensor (documentation, issues)
First occurred: 7:12:24 PM (2 occurrences)
Last logged: 7:12:24 PM

Error adding entities for domain sensor with platform nordpool
Error while setting up nordpool platform for sensor
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/template.py", line 407, in async_render
    render_result = _render_with_context(self.template, compiled, **kwargs)
  File "/usr/src/homeassistant/homeassistant/helpers/template.py", line 1820, in _render_with_context
    return template.render(**kwargs)
  File "/usr/local/lib/python3.9/site-packages/jinja2/environment.py", line 1291, in render
    self.environment.handle_exception()
  File "/usr/local/lib/python3.9/site-packages/jinja2/environment.py", line 926, in handle_exception
    raise rewrite_traceback_stack(source=source)
  File "<template>", line 1, in top-level template code
jinja2.exceptions.UndefinedError: 'dict object' has no attribute 'night'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 382, in async_add_entities
    await asyncio.gather(*tasks)
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 614, in _async_add_entity
    await entity.add_to_platform_finish()
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 798, in add_to_platform_finish
    await self.async_added_to_hass()
  File "/config/custom_components/nordpool/sensor.py", line 484, in async_added_to_hass
    await self.check_stuff()
  File "/config/custom_components/nordpool/sensor.py", line 445, in check_stuff
    self._update(today)
  File "/config/custom_components/nordpool/sensor.py", line 304, in _update
    formatted_prices = [
  File "/config/custom_components/nordpool/sensor.py", line 305, in <listcomp>
    self._calc_price(
  File "/config/custom_components/nordpool/sensor.py", line 268, in _calc_price
    template_value = self._ad_template.async_render(now=faker())
  File "/usr/src/homeassistant/homeassistant/helpers/template.py", line 409, in async_render
    raise TemplateError(err) from err
homeassistant.exceptions.TemplateError: UndefinedError: 'dict object' has no attribute 'night'

```

**Comment by Hellowlol** [comment]
Hi Vegard!

As I mention on Facebook you need to fix your template. The problem is that you don't have attributes night in the template (as the traceback reports) You do however have a winter_night in your template. I assume you have changed from winter_night to night in the template unintentionally.

You need to change ```{{s.night+s.hourly_fixed_cost+s.cert|float}}``` to ```{{s.winter_night+s.hourly_fixed_cost+s.cert|float}}```

**Comment by kjbu** [comment]
Hi. Thanks! This is not Vegard so it looks like several have the same problem. I did not change the template, was not logged in for several days so this appeared suddenly. 

I have downloaded the template from somewhere, probably with the error, several months ago. 

Changing the code as you mentioned fixed the sensor 👍

I will keep this thread here to help others. 
My nordpool apexchart don't work now after update to apex v.2.0.1 so that is my next step to figure out of 🙈
